jetty-io/src/main/java/org/eclipse/jetty/io/ssl/SslConnection.java

```diff
@@ -866,7 +870,7 @@ public boolean flush(ByteBuffer... appOuts) throws IOException
                                         if (filled < 0)
                                             throw new IOException("Broken pipe");
                                     }
-                                    return result = false;
+                                    return result = appOuts.length==1 && BufferUtil.isEmpty(appOuts[0]);
```

Comment:
I'm not sure about this.
It may happen that we do wrap and consume the `appOuts`, but we have unflushed bytes in `_encryptedOutput`.
Also the special case of testing for `appOuts.length==1` does not seem justified.
Here we are flushing, we have been told to unwrap, we filled but was not enough, I think we should just return `false` - if we return `true` we may never finish to unwrap.

Reply:
The true false return only indicates if we are flushed or not.    
If we have no encrypted bytes, no unencrypted bytes, then we are flushed even if we are in NEED_UNWRAP.   I'll update to reflect that better with less special case pleading....